db/blip_handler.go

```diff
@@ -365,6 +365,29 @@ func (bh *blipHandler) handleSubChanges(rq *blip.Message) error {
 		base.DebugfCtx(bh.loggingCtx, base.KeySyncMsg, "#%d: Type:%s   --> Time:%v", bh.serialNumber, rq.Profile(), time.Since(startTime))
 	}()
 
+	auditFields := base.AuditFields{
+		base.AuditFieldSince: subChangesParams.Since().String(),
+	}
+	var filters []string
+	if len(subChangesParams.docIDs()) > 0 {
+		auditFields[base.AuditFieldDocIDs] = subChangesParams.docIDs()
+		filters = append(filters, base.DocIDsFilter)
+	}
+	if subChangesParams.filter() != "" {
+		filters = append(filters, subChangesParams.filter())
+	}
+	if len(filters) > 0 {
+		auditFields[base.AuditFieldFilter] = strings.Join(filters, ",")
```

Comment:
We don't support multiple filters for a changes request.  We can filter either by channels (base.ByChannelsFilter) or by doc ids (base.DocIDsFilter).  If it's the former, we also set the channels property - if it's the latter we set docIDs.

Reply:
We discussed a bit offline and I worry that it is not actually enforced by CBL protocol - 

https://github.com/couchbase/couchbase-lite-core/blob/master/modules/docs/pages/replication-protocol.adoc#subchanges

The cbl code also doesn't enforce this (at this level): https://github.com/couchbase/couchbase-lite-core/blob/bc77f743d13fb6de86e56167425cdb3252ae7c71/Replicator/Puller.cc#L82

It can only set filter to be `sync_gateway/bychannel` but it looks like it independently passes docids here. For REST, these are exclusive.

In practice, lite may never access this code pathway, or it might be validated elsewhere.

Reply:
In GenerateChanges you can see that if docIDs are specified, it's going to use DocIDChangesFeed.  And DocIDChangesFeed doesn't do anything with channels, even if CBL decided to go rogue and send both.  

So from an auditing perspective, I would audit the filter as DocIDs if DocIDs are present, as that's going to be the Sync Gateway behaviour.